### ISSUE TITLE
Checkout Button Missing after upgrade to 3.3.1

### ISSUE THREAD

**Issue by JMIANO1981** [woocommerce/woocommerce#18870]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [x ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. After the update, the checkout button becomes missing.
2. I rollback the update and still the checkout button still missing.
3. I update the theme and still, the woocommerce plugin have an issue with checkout button and the photo of the item on the right side also missing.

## Expected/actual behavior

When I follow those steps, I see...

I was expecting to see...

## Isolating the problem

- [x ] This bug happens with only WooCommerce plugin active
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [ ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: http://www.sabangseaferrympc.com/wp
Site URL: http://www.sabangseaferrympc.com/wp
WC Version: 3.3.1
Log Directory Writable: ✔
WP Version: 4.9.4
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.1.6
PHP Post Max Size: 128 MB
PHP Time Limit: 180
PHP Max Input Vars: 5000
cURL Version: 7.54.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.35
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.1
WC Database Prefix: wp_
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.
Total Database Size: 16.44MB
Database Data Size: 14.09MB
Database Index Size: 2.35MB
wp_woocommerce_sessions: Data: 0.01MB + Index: 0.00MB
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.04MB
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_apmm_custom_theme: Data: 0.00MB + Index: 0.00MB
wp_commentmeta: Data: 0.31MB + Index: 0.12MB
wp_comments: Data: 0.34MB + Index: 0.06MB
wp_content_tabs_ultimate_import: Data: 0.00MB + Index: 0.00MB
wp_content_tabs_ultimate_list: Data: 0.01MB + Index: 0.00MB
wp_content_tabs_ultimate_style: Data: 0.00MB + Index: 0.00MB
wp_duplicator_packages: Data: 0.01MB + Index: 0.00MB
wp_es_deliverreport: Data: 0.01MB + Index: 0.00MB
wp_es_emaillist: Data: 0.00MB + Index: 0.00MB
wp_es_notification: Data: 0.00MB + Index: 0.00MB
wp_es_sentdetails: Data: 0.04MB + Index: 0.00MB
wp_failed_jobs: Data: 0.00MB + Index: 0.00MB
wp_hotel_room: Data: 0.00MB + Index: 0.00MB
wp_huge_itgallery_gallerys: Data: 0.00MB + Index: 0.00MB
wp_huge_itgallery_images: Data: 0.00MB + Index: 0.00MB
wp_huge_itgallery_like_dislike: Data: 0.00MB + Index: 0.00MB
wp_iowd_images: Data: 0.00MB + Index: 0.00MB
wp_links: Data: 0.00MB + Index: 0.00MB
wp_mailchimp_carts: Data: 0.00MB + Index: 0.00MB
wp_mwp_side_menu_free: Data: 0.00MB + Index: 0.00MB
wp_options: Data: 2.87MB + Index: 0.12MB
wp_postmeta: Data: 5.14MB + Index: 1.51MB
wp_posts: Data: 4.74MB + Index: 0.22MB
wp_queue: Data: 0.00MB + Index: 0.00MB
wp_st_activity: Data: 0.00MB + Index: 0.00MB
wp_st_availability: Data: 0.02MB + Index: 0.01MB
wp_st_cars: Data: 0.00MB + Index: 0.00MB
wp_st_cba_statistic: Data: 0.00MB + Index: 0.00MB
wp_st_flights: Data: 0.00MB + Index: 0.00MB
wp_st_flight_airport: Data: 0.00MB + Index: 0.00MB
wp_st_flight_availability: Data: 0.25MB + Index: 0.06MB
wp_st_flight_location: Data: 0.00MB + Index: 0.00MB
wp_st_form_builder: Data: 0.01MB + Index: 0.00MB
wp_st_hotel: Data: 0.00MB + Index: 0.00MB
wp_st_inbox: Data: 0.00MB + Index: 0.00MB
wp_st_journey_car: Data: 0.00MB + Index: 0.00MB
wp_st_location_nested: Data: 0.00MB + Index: 0.00MB
wp_st_location_relationships: Data: 0.02MB + Index: 0.01MB
wp_st_member_packages: Data: 0.00MB + Index: 0.00MB
wp_st_member_packages_order: Data: 0.00MB + Index: 0.00MB
wp_st_order_item_meta: Data: 0.04MB + Index: 0.00MB
wp_st_price: Data: 0.00MB + Index: 0.00MB
wp_st_properties: Data: 0.01MB + Index: 0.00MB
wp_st_rental: Data: 0.00MB + Index: 0.00MB
wp_st_tours: Data: 0.00MB + Index: 0.00MB
wp_st_user_online: Data: 0.06MB + Index: 0.02MB
wp_st_withdrawal: Data: 0.00MB + Index: 0.00MB
wp_termmeta: Data: 0.00MB + Index: 0.01MB
wp_terms: Data: 0.01MB + Index: 0.02MB
wp_term_relationships: Data: 0.04MB + Index: 0.08MB
wp_term_taxonomy: Data: 0.01MB + Index: 0.02MB
wp_tp_auto_replac_links: Data: 0.00MB + Index: 0.00MB
wp_tp_hotel_list_shortcode: Data: 0.00MB + Index: 0.00MB
wp_tp_search_shortcodes: Data: 0.00MB + Index: 0.00MB
wp_usermeta: Data: 0.04MB + Index: 0.02MB
wp_users: Data: 0.00MB + Index: 0.01MB
wp_visual_form_builder_entries: Data: 0.00MB + Index: 0.00MB
wp_visual_form_builder_fields: Data: 0.00MB + Index: 0.00MB
wp_visual_form_builder_forms: Data: 0.00MB + Index: 0.00MB
wp_wcpdf_invoice_number: Data: 0.00MB + Index: 0.00MB
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB
wp_wdslayer: Data: 0.00MB + Index: 0.00MB
wp_wdsslide: Data: 0.01MB + Index: 0.00MB
wp_wdsslider: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_buckaroo_transactions: Data: 0.00MB + Index: 0.00MB
wp_wsluserscontacts: Data: 0.00MB + Index: 0.00MB
wp_wslusersprofiles: Data: 0.00MB + Index: 0.00MB
wp_yoast_seo_links: Data: 0.03MB + Index: 0.01MB
wp_yoast_seo_meta: Data: 0.00MB + Index: 0.00MB

### Post Type Counts ###

attachment: 464
hotel_room: 45
jetpack_migration: 2
jp_img_sitemap: 3
jp_sitemap: 3
jp_sitemap_master: 3
location: 15
mc4wp-form: 1
nav_menu_item: 220
option-tree: 1
page: 150
post: 12
product: 94
product_variation: 98
rental_room: 10
revision: 465
shop_order: 28
st_activity: 15
st_cars: 32
st_flight: 11
st_hotel: 29
st_layouts: 45
st_mega_menu: 1
st_order: 24
st_rental: 20
st_tours: 72
tour: 1
vcv_templates: 9
wds-slider: 1
wpcf7_contact_form: 6
wpforms: 1

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (38) ###

Akismet Anti-Spam: by Automattic – 4.0.2
Contact Form 7 Conditional Fields: by Jules Colle – 1.3.4
Contact Form 7 Datepicker: by Aurel Canciu – 2.6.0
Contact Form 7: by Takayuki Miyoshi – 5.0
cruise-common: by Vergatheme – 1.0
Duplicate Page: by mndpsingh287 – 2.5
Events Manager Pro - Bank Transfer gateway: by Florent Maillefaud – 0.1
GA Google Analytics: by Jeff Starr – 20171103
Huge IT Image Gallery: by Huge-IT – 2.2.6
Google Analytics Dashboard for WP (GADWP): by Alin Marcu – 5.1.2.5
Hello Dolly: by Matt Mullenweg – 1.6
Hide Page And Post Title: by Arjun Thakur – 1.1
Image Optimizer WD: by WebDorado – 1.0.9
Jetpack by WordPress.com: by Automattic – 5.8
WPBakery Page Builder: by Michael M - WPBakery.com – 5.4.5
Limit Login Attempts: by Johan Eenfeldt – 1.7.1
MailChimp for WordPress: by ibericode – 4.1.15
Wow Side Menu: by Wow-Company – 3.1.1
OptionTree: by Derek Herman – 2.6.0
Reset WP: by Nikunj Soni – 1.1
ShortPixel Image Optimizer: by ShortPixel – 4.9.0
Slider WD: by WebDorado – 1.2.5
Smartsupp Live Chat: by Smartsupp – 3.2 – Not tested with the active version of WooCommerce
Terms and Conditions Popup for WooCommerce: by BeRocket – 1.0.8 – Not tested with the active version of WooCommerce
Traveler-code: by Shinetheme – 2.0.3
Visual Form Builder: by Matthew Muro – 2.9.4
W3 Total Cache: by Frederick Townes – 0.9.6
WooCommerce Checkout Manager: by Visser Labs – 4.1.6 – Not tested with the active version of WooCommerce
WooCommerce Extra Charges To Payment Gateway (Standard): by hemsingh1 – 1.0.12.9 – Not tested with the active version of WooCommerce
WooCommerce PayPal Express Checkout Gateway: by WooCommerce – 1.5.1
WooCommerce PDF Invoices & Packing Slips: by Ewout Fernhout – 2.1.4
WooCommerce: by Automattic – 3.3.1
WordPress Importer: by wordpressdotorg – 0.6.3
Yoast SEO: by Team Yoast – 6.2
WordPress Social Login: by Miled – 2.3.3
WP-Optimize: by David Anderson
Ruhani Rabin
Team Updraft – 2.1.1

WPForms Lite: by WPForms – 1.4.4
YITH WooCommerce PDF Invoice and Shipping List: by YITHEMES – 1.2.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: PHP (₱)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #5189 - /shop/
Cart: #5190 - /cart/
Checkout: #5191 - /checkout-woo-booking/
My account: ❌ Page does not contain the shortcode.
Terms and conditions: #7661 - /terms-and-conditions/

### Theme ###

Name: Traveler
Version: 2.0.3
Author URL: http://www.shinetheme.com
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: traveler/woocommerce/archive-product.php version 2.0.0 is out of date. The core version is 3.3.0
traveler/woocommerce/cart/cart-empty.php
traveler/woocommerce/cart/cart.php version 3.1.0 is out of date. The core version is 3.3.0
traveler/woocommerce/checkout/form-checkout.php
traveler/woocommerce/checkout/form-coupon.php version 2.2 is out of date. The core version is 3.3.0
traveler/woocommerce/checkout/review-order.php version 2.3.0 is out of date. The core version is 3.3.0
traveler/woocommerce/checkout/thankyou.php
traveler/woocommerce/content-product.php
traveler/woocommerce/content-single-product.php
traveler/woocommerce/content-widget-product.php version 2.5.0 is out of date. The core version is 3.3.0
traveler/woocommerce/emails/admin-new-order.php
traveler/woocommerce/global/breadcrumb.php
traveler/woocommerce/global/sidebar.php
traveler/woocommerce/loop/orderby.php version 2.2.0 is out of date. The core version is 3.3.0
traveler/woocommerce/loop/pagination.php version 2.2.2 is out of date. The core version is 3.3.1
traveler/woocommerce/loop/sale-flash.php
traveler/woocommerce/order/order-details.php version 3.2.0 is out of date. The core version is 3.3.0
traveler/woocommerce/product-searchform.php version 2.5.0 is out of date. The core version is 3.3.0
traveler/woocommerce/single-product/sale-flash.php
traveler/woocommerce/single-product/tabs/additional-information.php
traveler/woocommerce/single-product/tabs/description.php
traveler/woocommerce/single-product-reviews.php

Outdated Templates: ❌Learn how to update
`
```
</details>

**Comment by WPprodigy** [comment]
Duplicating an issue isn't going return different result: https://github.com/woocommerce/woocommerce/issues/18865#issuecomment-364024034

I'll leave this one open for you though, so you can hear the same from somebody else ¯\\\_(ツ)\_/¯

**Comment by kloon** [comment]
@JMIANO1981 please include a systems status report and follow the steps as @WPprodigy mentioned in the other issue. There is not much we can do to help you fix this if we do not have the right information to work with.

The issue is most likely with your theme which overrides core WooCommerce templates, so please following the advice about disabling all plugins except WooCommerce and switching to a default theme like Storefront or Twentyseventeen.

**Comment by JMIANO1981** [comment]
@kloon I did already in the details.

**Comment by kloon** [comment]
OK, so with that I can see your theme is overriding core WooCommerce templates, specifically on the checkout pages.

I'd suggest you test by switching temporarily to a default theme like Storefront or Twentyseventeen and then also disable all plugins except WooCommerce and test this again.

If I test this with WooCommerce core on a default theme the button is there and works.

**Comment by JMIANO1981** [comment]
@kloon I did on my other website (personal) and I also encountered the same issues.

**Comment by mikejolley** [comment]
I am seeing syntax errors coming from media modals/upload code on your page. This is not WooCommerce.

I am seeing a notice in the header from a hide post title plugin. This is not WooCommerce.

> I rollback the update and still the checkout button still missing.

Which shows it's not a problem with the update of WooCommerce.

This is the source of your checkout page:

![checkout for woo booking - sabang sea ferry mpc 2018-02-08 11-29-38](https://user-images.githubusercontent.com/90977/35970754-6c6a592a-0cc3-11e8-8758-cb87ce1f9822.png)

Which shows:

1. you have custom template overrides changing the markup
2. Someone has hidden the terms and conditions box with a display:none rule
3. Because terms is hidden, the checkout button (which is there) is also hidden.

TL;DR this is a theme issue which you would have found by going through our debug steps correctly. Whomever added the display:none on the terms and conditions area is to blame. This is not due to the update at all.

**Comment by JMIANO1981** [comment]
Hello Mike Jolly,

I extracted the zip file of WooCommerce 3.3.1 and to the checkout folder
which the terms.php and it also shows the display: none which the same.

I also extracted the lower version which 3.3 and also the same.

Here is from the 3.3.1
echo '<div class="woocommerce-terms-and-conditions" style="display: none;
max-height: 200px; overflow: auto;">

I also check from github
https://github.com/woocommerce/woocommerce/blob/master/templates/checkout/terms.php
:

Which shows the same.

I made the terms appeared on display: auto but still unclickable.

I already speak with Theme Vendor and they've said, there should be no
conflict with the updates unless the woocommerce itself has a problem with
a bug.



[image: photo]
Jonathan Miano, Jr
Website Developer | IT Consultancy | General VA at  JMIANO-VA
A  167 (B) 5th Ave. Riverside Village, Pasig City, Philippines
P  (+63947) 388 0650  <(+63947) 388 0650>
M  (+63927) 853 6199  <(+63927) 853 6199>
E  JBMJR243@gmail.com  <JBMJR243@gmail.com>
W  www.jmiano-va.com  <http://www.jmiano-va.com>
Skype JBMJR243@gmail.com <#>
<http://www.facebook.com/JMIANO243>
<http://www.linkedin.com/in/jonathan-jr-miano-95330158/>
<http://twitter.com/jmianojr>

Please consider your environmental responsibility. Before printing this
e-mail message, ask yourself whether you really need a hard copy.
Create your own email signature
<https://wisestamp.com/email-install?utm_source=promotion&utm_medium=signature&utm_campaign=create_your_own>

**Comment by mikejolley** [comment]
If you look here, the place order button is not within the terms area.

https://github.com/woocommerce/woocommerce/blob/master/templates/checkout/payment.php#L51

This should be ample information for your theme to resolve.

**Comment by JMIANO1981** [comment]
Hey Mike no need to worry, I figure it out myself.  Other plugins have
nothing to do, the code itself has some errors or bugs that mixes up the
output.

Thank you by the way.


[image: photo]
Jonathan Miano, Jr
Website Developer | IT Consultancy | General VA at  JMIANO-VA
A  167 (B) 5th Ave. Riverside Village, Pasig City, Philippines
P  (+63947) 388 0650  <(+63947) 388 0650>
M  (+63927) 853 6199  <(+63927) 853 6199>
E  JBMJR243@gmail.com  <JBMJR243@gmail.com>
W  www.jmiano-va.com  <http://www.jmiano-va.com>
Skype JBMJR243@gmail.com <#>
<http://www.facebook.com/JMIANO243>
<http://www.linkedin.com/in/jonathan-jr-miano-95330158/>
<http://twitter.com/jmianojr>

Please consider your environmental responsibility. Before printing this
e-mail message, ask yourself whether you really need a hard copy.
Create your own email signature
<https://wisestamp.com/email-install?utm_source=promotion&utm_medium=signature&utm_campaign=create_your_own>
